BootloaderCorePkg/Tools/CommonUtility.py

```diff
@@ -275,8 +275,13 @@ def rsa_sign_file (priv_key, pub_key, hash_type, sign_scheme, in_file, out_file,
     if len(bins) != len(out_data):
         gen_file_from_object (out_file, bins)
 
-def get_key_type (in_key):
-    pub_key = gen_pub_key (in_key)
+def get_key_type (in_priv_key, in_pub_key = None):

```

Comment:
how about using get_key_type (in_key) ?
in_key can be  PEM  private or public key file or binary file for public key.
if in_key.endswith('.pem') then  do  gen_pub_key(in_key),  else do the file read.

Reply:
I think there should not be restriction for private key file end with pem format.
Alternatively we can maintain separate function getting key size from public key

Reply:
in_priv_key is not accurate since it could be private or public key. both are supported.

Reply:
Thanks. Added check Public key identifier for public key in bin format.
Updated PR.